src/vswhere.lib/JsonFormatter.cpp

```diff
@@ -26,49 +29,49 @@ wstring JsonFormatter::Escape(_In_ const wstring& value)
     return buffer;
 }
 
-void JsonFormatter::StartArray(_In_ Console& console, _In_opt_ const std::wstring& name)
+void JsonFormatter::StartArray(_In_opt_ const std::wstring& name)
 {
-    StartScope(console, JsonScope::Type::array, name);
+    StartScope(JsonScope::Type::array, name);
 }
 
-void JsonFormatter::StartObject(_In_ Console& console, _In_opt_ const wstring& name)
+void JsonFormatter::StartObject(_In_opt_ const wstring& name)
 {
-    StartScope(console, JsonScope::Type::object, name);
+    StartScope(JsonScope::Type::object, name);
 }
 
-void JsonFormatter::WriteProperty(_In_ Console& console, _In_ const wstring& name, _In_ const wstring& value)
+void JsonFormatter::WriteProperty(_In_ const wstring& name, _In_ const wstring& value)
 {
-    StartProperty(console, name);
+    StartProperty(name);
 
     auto escaped = Escape(value);
-    console.Write(L"\"%ls\"",escaped.c_str());
+    Console().Write(L"%ls\"%ls\"%ls", Console().Color(ColorValue), escaped.c_str(), Console().ResetColor());
```

Comment:
Might be helpful to create a helper for this since we repeat this set, print, reset pattern throughout

Reply:
These are helpers. :) They emit VT100-ish sequences if color is enabled or an empty string.

If you abstract it away much more, you start running into unnecessary compexities. I went through several iterations and this turned out to be the cleanest and leanest.